clients/src/main/java/org/apache/kafka/clients/consumer/internals/events/ApplicationEventProcessor.java

```diff
@@ -234,7 +239,10 @@ private void process(final AsyncCommitEvent event) {
 
         try {
             CommitRequestManager manager = requestManagers.commitRequestManager.get();
-            CompletableFuture<Map<TopicPartition, OffsetAndMetadata>> future = manager.commitAsync(event.offsets());
+            Optional<Map<TopicPartition, OffsetAndMetadata>> offsets = event.offsets().isEmpty() ?
+                Optional.of(subscriptions.allConsumed()) : event.offsets();
+            event.markOffsetsReady();
```

Comment:
I introduce the logic here to see if we need to invoke `allCounsumed`. After that, it will complete the `offsetsReady` to notify the app thread the offsets are up to date.

It's quite nasty here; I will improve it if we agree on this solution.